code/modules/reagents/reagent_containers/food/snacks.dm

```diff
@@ -4571,6 +4571,81 @@
 	LEXORIN=5, GRAVY=5, DETCOFFEE=5, AMUTATIONTOXIN=5, GYRO=5, SILENCER= 5, URANIUM=5)
 	var/reagent=pick(possible_reagents)
 	reagents.add_reagent(reagent, possible_reagents[reagent])
+	
+/obj/item/weapon/reagent_containers/food/snacks/lollipop
+	name = "lollipop"
+	desc = "Suck on this!"
+	icon_state = "lollipop_stick"
+	item_state = "lollipop_stick"
+	food_flags = FOOD_SWEET
+	icon = 'icons/obj/candymachine.dmi'
+	bitesize = 2
+	slot_flags = SLOT_MASK //No, really, suck on this.
+	attack_verb = list("taps", "pokes")
+	eatverb = "crunch"
+	edible_by_utensil = FALSE
+	trash = /obj/item/trash/lollipopstick
+	var/candyness = 100 //how long this thing will last
+	volume = 15 //not a lotta room for poison
+	
+
+/obj/item/weapon/reagent_containers/food/snacks/lollipop/New()
+	..() 
+	eatverb = pick("bite","crunch","chomp")
+	reagents.add_reagent(NUTRIMENT, 2)
+	reagents.add_reagent(SUGAR, 8)
+	var/list/random_color_list = list("#00aedb","#a200ff","#f47835","#d41243","#d11141","#00b159","#00aedb","#f37735","#ffc425","#008744","#0057e7","#d62d20","#ffa700")
+	var/image/colorpop = image('icons/obj/candymachine.dmi', icon_state = "lollipop_head")
+	colorpop.color = pick(random_color_list)
+	src.overlays += colorpop
+	filling_color = colorpop.color
+	
+/obj/item/weapon/reagent_containers/food/snacks/lollipop/consume()
+	..()
+	candyness -= bitesize*10 //taking a bite out reduces how long it'll last, duh
+	
+/obj/item/weapon/reagent_containers/food/snacks/lollipop/proc/updateconsuming(var/sucking)
+	if(sucking)
+		processing_objects.Add(src)
+	else
+		processing_objects.Remove(src)		
+
+/obj/item/weapon/reagent_containers/food/snacks/lollipop/Destroy()
+	. = ..()
+
+	processing_objects -= src
+
+/obj/item/weapon/reagent_containers/food/snacks/lollipop/process()
+	var/mob/living/carbon/human/H = get_holder_of_type(src,/mob/living/carbon/human)
+	if(H.get_item_by_slot(slot_wear_mask) == src)
+		candyness--
+	if(candyness <= 0)
+		processing_objects.Remove(src)
+		var/turf/T = get_turf(src)
+		var/atom/new_stick = new /obj/item/trash/lollipopstick(T)
+		transfer_fingerprints_to(new_stick)
+		H.equip_to_slot(new_stick, slot_wear_mask, 1)
```

Comment:
why are you doing this before deleting the lollipop? won't it fail?
why are you telling it not to redraw the mask and then redrawing the mask manually immediately after?

Reply:
It won't fail because I'm using the unsafe proc, essentially overwriting whatever's in the slot. Could swap it for sanity's sake, if necessary.
Far as I know, one is for the mob sprite, and one is for the inventory sprite. The mob sprite doesn't (need to) change, the inventory one does.